pkg/analyzer/analyzer.go

```diff
@@ -694,18 +694,19 @@ func isConfigFile(path string, exc []string) bool {
 }
 
 // shouldConsiderGitIgnoreFile verifies if the scan should exclude the files according to the .gitignore file
-func shouldConsiderGitIgnoreFile(path, gitIgnore string, excludeGitIgnoreFile bool) (bool, *ignore.GitIgnore) {
+func shouldConsiderGitIgnoreFile(path, gitIgnore string, excludeGitIgnoreFile bool) (hasGitIgnoreFileRes bool,
+	gitIgnoreRes *ignore.GitIgnore, pathRes string) {
 	gitIgnorePath := filepath.ToSlash(filepath.Join(path, gitIgnore))
 	_, err := os.Stat(gitIgnorePath)
 
 	if !excludeGitIgnoreFile && err == nil && gitIgnore != "" {
 		gitIgnore, _ := ignore.CompileIgnoreFile(gitIgnorePath)
 		if gitIgnore != nil {
 			log.Info().Msgf(".gitignore file was found in '%s' and it will be used to automatically exclude paths", path)
-			return true, gitIgnore
+			return true, gitIgnore, path
```

Comment:
why is the variable returned the same as the first parameter? is this the correct variable to return?

Reply:
I added an E2E test and fixed the variable, thanks for the suggestions!

Reply:
https://github.com/Checkmarx/kics/pull/6896/commits/8ac4222fe8a2595377523d2a363894549dccce6a